plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -326,7 +327,7 @@ bool GazeboYarpControlBoardDriver::gazebo_init()
             }
             else if (coupling_bottle->get(0).asString()=="icub_hand_mk3")
             {
-                BaseCouplingHandler* cpl = new HandMk3CouplingHandler(m_robot,coupled_joints, coupled_joint_names, coupled_joint_limits);
+                BaseCouplingHandler* cpl = new HandMk4CouplingHandler(m_robot,coupled_joints, coupled_joint_names, coupled_joint_limits);
```

Comment:
This piece of code uses the `HandMk4CouplingHandler` class if the coupling name asked is `icub_hand_mk3`, I am afraid this may be confusing, what do you think? See also the subsquent message `yCInfo(GAZEBOCONTROLBOARD) << "using icub_hand_mk3";`. Perhaps we could have a specific coupling name for this?

Reply:
I kept the naming `icub_hand_mk3` to be back-compatible on the user-configuration level, but I agree, it might be better to give it a specific name and separate things and maybe keep both classes for the time being.

Reply:
Yes, I think in any case you would not be able to be back-compatible as the number of joints is different, right?